render-test/runner.cpp

```diff
@@ -383,10 +404,83 @@ bool TestRunner::checkRenderTestResults(mbgl::PremultipliedImage&& actualImage,
             return false;
         }
     }
+    // Check gfx metrics
+    for (const auto& expected : metadata.expectedMetrics.gfx) {
+        auto actual = metadata.metrics.gfx.find(expected.first);
+        if (actual == metadata.metrics.gfx.end()) {
+            metadata.errorMessage = "Failed to find gfx probe: " + expected.first;
+            return false;
+        }
+
+        auto check = [&metadata](
+                         const char* msg, const std::string& name, const auto& expected, const auto& actual) -> bool {
+            if (expected != actual) {
+                char msgBuf[512];
+                snprintf(msgBuf, sizeof(msgBuf), msg, name.c_str(), actual, expected);
+                metadata.errorMessage = msgBuf;
+                return false;
+            }
+            return true;
+        };
+
+        const auto& probeName = expected.first;
+        const auto& expectedValue = expected.second;
+        const auto& actualValue = actual->second;
+
+        if (!check("Number of draw calls at probe \"%s\" is %i, expected is %i",
```

Comment:
I think it would be better to check all probes and return 'false' if one of probe fails. Also, are all expected values mandatory?

Reply:
Very good point!